### PR TITLE
feat(manager/npm): ignore scripts for `pnpmDedupe`

### PR DESCRIPTION
**Pull request by liby**
## Changes

`pnpmDedupe` will not run the lifecycle script.

## Context

https://github.com/renovatebot/renovate/discussions/24417#discussioncomment-7004299

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
I don't think a new option is needed here. The pnpm dedupe should follow the existing logic for whether to ignore scripts or not

**Comment by liby** [comment]
> I don't think a new option is needed here.

The reason for doing this is to refer to the suggestions in section "[Adding configuration options](https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md#adding-configuration-options)".

However, I agree with your point, and I will make the necessary changes.

Of course, this PR cannot be merged yet, as pnpm has not yet released a version that includes the `pnpm dedupe --ignore-scripts`, so there's no rush.

**Comment by rarkins** [comment]
@liby congratulations on getting https://github.com/pnpm/pnpm/pull/7103 merged into pnpm@8.8.0

Can you now resume this PR?

First question is: does pnpm <8.8.0 gracefully ignore any `--ignore-scripts`? If so then we can apply it to all. If not then we need to _selectively_ apply it somehow.

**Comment by liby** [comment]
> First question is: does pnpm <8.8.0 gracefully ignore any --ignore-scripts?

No, passing `--ignore-scripts` in versions below pnpm@8.8.0 will cause an error.
```
❯ pnpm dedupe --ignore-scripts
 ERROR   ERROR  Unknown option: 'ignore-scripts'
Did you mean 'ignore-workspace'? Use "--config.unknown=value" to force an unknown option.
For help, run: pnpm help dedupe
```

>  If not then we need to selectively apply it somehow.

[Adds a check for the pnpm version before passing `ignore-scripts`](https://github.com/renovatebot/renovate/pull/24505/commits/d680d25018b75f3d655f01e5c9cb5eff07691af1). It's not the most elegant solution, but it gets the job done for now.

Do you have any better ideas or other solutions?

File: lib/modules/manager/npm/post-update/pnpm.ts [hunk 1]
```diff
@@ -77,10 +79,14 @@ export async function generateLockFile(
     logger.trace({ cmd, args }, 'pnpm command');
     commands.push(`${cmd} ${args}`);
 
+    const pnpmVersion = execSync('pnpm --version').toString().trim();
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/pnpm.ts:82]: It might not be installed by here so this won't work reliably

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/npm/post-update/pnpm.ts (hunk 1 above)
Comment (liby) [reply lib/modules/manager/npm/post-update/pnpm.ts]: Considering your point about `pnpm` potentially not being installed at this stage, I propose using the existing `getPackageManagerVersion` method to retrieve the `pnpm` version from the `package.json`. This method should provide us with a more robust mechanism.

In cases where the `engines` field is missing in the `package.json`, or when we can't get a valid version for other reasons, I plan to not pass the `--ignore-scripts` option when executing `pnpm dedupe`.

Let me know your thoughts on this approach.

**Review by rarkins — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**